conda_forge_tick/make_graph.py

```diff
@@ -0,0 +1,107 @@
+import codecs
+import datetime
+import hashlib
+import os
+import re
+import time
+from collections import defaultdict
+import networkx as nx
+import requests
+import jinja2
+from .utils import parsed_meta_yaml
+from .all_feedstocks import get_all_feedstocks
+
+import logging
+
+logger = logging.getLogger("conda_forge_tick.make_graph")
+
+def make_graph(names, gx=None):
+
+    bad = []
+
+    logger.info('reading graph')
+
+    if gx is None:
+        gx = nx.DiGraph()
+
+    new_names = [name for name in names if name not in gx.nodes]
+    old_names = [name for name in names if name in gx.nodes]
+    old_names = sorted(old_names, key=lambda n: gx.nodes[n]['time'])
+
+    total_names = new_names + old_names
+    logger.info('start loop')
+
+    for i, name in enumerate(total_names):
+        logger.info((i, name))
+        r = requests.get('https://raw.githubusercontent.com/conda-forge/'
+                         '{}-feedstock/master/recipe/meta.yaml'.format(name))
+        if r.status_code != 200:
+            logger.warn('Something odd happened when fetching recipe {}: '
+                  '{}'.format(name, r.status_code))
+            bad.append(name)
+            continue
+
+        text = r.content.decode('utf-8')
+        yaml_dict = parsed_meta_yaml(text)
+        if not yaml_dict:
+            logger.warn('Something odd happened when parsing recipe {}'.format(name))
+            bad.append(name)
+            continue
+        # TODO: Write schema for dict
+        req = yaml_dict.get('requirements', set())
+        if req:
+            build = list(req.get('build', []) if req.get(
+                'build', []) is not None else [])
+            run = list(req.get('run', []) if req.get(
+                'run', []) is not None else [])
+            req = build + run
+            req = set([x.split()[0] for x in req])
+
+        if not ('url' in yaml_dict.get('source', {})
+                and 'name' in yaml_dict.get('package', {})
+                and 'version' in yaml_dict.get('package', {})):
+            logger.warn("Recipe {} doesn't have a url".format(name))
```

Comment:
Here the problem was that the recipe was missing url, name, or version, but we do not know which after the if statement.